src/main/java/org/labellum/mc/dynamictreestfc/DTTFCGenerator.java

```diff
@@ -68,56 +51,49 @@ public boolean canGenerateTree(World world, BlockPos pos, Tree treeType)
             return false;
         }
 
-        if (radiusCoordinator == null)
-        {
-            radiusCoordinator = new TFCRadiusCoordinator(null,world);
-        }
-
-        dtRadius = radiusCoordinator.getRadiusAtCoords(pos.getX(),pos.getZ());
-
-        //check on ground and nearby trees
-        int x;
-        int z;
         Species dTree = ModTrees.tfcSpecies.get(treeType.toString());
-        int nht = dTree.getLowestBranchHeight();
-        int xht = (int)((TreeFamilyTFC.TreeTFCSpecies)dTree).getSignalEnergy(); //signal energy access problem so need to cast
+        int lowestBranchHeight = dTree.getLowestBranchHeight();
+        int maxTreeHeight = (int) ((TreeFamilyTFC.TreeTFCSpecies) dTree).getSignalEnergy(); //signal energy access problem so need to cast
 
         SafeChunkBounds bounds = new SafeChunkBounds(world, world.getChunk(pos).getPos());
-        for ( int y = 0; y <= nht; y++) {
-            if (openRadius(world,pos.up(y),0,0, bounds))
-            {
-                continue;
-            }
-            return false;
+        for (int y = 0; y <= lowestBranchHeight; y++)
+        {
+            if (!isValidLocation(world, pos.up(y), 0, 0, bounds))
+                return false; // ensure proper column space for the trunk
         }
+        leavesRadius = (int) Math.ceil((1 - ChunkDataTFC.get(world, pos).getFloraDensity()) * 5) + 2; // what's a good proxy for a noise map? TFC gen data!
+        int radiusSquared = leavesRadius * leavesRadius;
+        int groundToCenter = (int) ((maxTreeHeight - lowestBranchHeight) / 2.0F) + lowestBranchHeight;
 
-        for(x = -dtRadius-1; x <= dtRadius+1; ++x) {
-            for(z = -dtRadius-1; z <= dtRadius+1; ++z) {
-                for ( int y = nht-1; y < xht; ++y)
+        for (int x = -leavesRadius - 1; x <= leavesRadius + 1; ++x) // verifying there's space for the canopy
+        {
+            for (int z = -leavesRadius - 1; z <= leavesRadius + 1; ++z)
+            {
+                for (int y = lowestBranchHeight - 1; y < maxTreeHeight; ++y)
                 {
-                    if (openRadius(world, pos.up(y), x, z, bounds))
+                    int yDistance = groundToCenter - y;
+                    if (x * x + yDistance * yDistance + z * z <= radiusSquared) // only perform the check if the radius is within a sphere around the epicenter there
```

Comment:
I thought you were talking about checking a cylinder. I don't think a sphere is going to work because the library of world gen used tree shapes do not match very well with the actual parameters of TFC trees, so we can't use these parameters to posit a tree shape. Many trees also don't have spherical growth habits period, and will have the farthest radius at the lowest height. 
Have you flown around a world with this generator and observed very few "frankentrees"? That's the acid test. Any time two DT have branches in adjacent blocks, the branch/trunk thickness algorithms freak out and make trees that are super ugly.

Reply:
well what the old algorithm does specifically is it tries to eliminate a big cube involving all of the leaves it expects to generate. To be honest we can just go back to that and it wouldn't be a big deal. The point of cancelling checks near the bottom or top is to make room for trees to be a bit more snug, that's all. If you dig around and see what values the radius thing actually takes, it used to only check 2-4 blocks out whereas now it checks much further out.

Flying around it seemed pretty much fine, not a lot of frankentrees as you say. But you're a better judge of what looks right than I.